.github/workflow/release.yml

```diff
@@ -0,0 +1,55 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release:
+    name: Release
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code repository
+        # https://github.com/actions/checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: setup node.js
+        # https://github.com/actions/setup-node
+        uses: actions/setup-node@v4
+
+      - name: install pnpm
+        ruses: pnpm/action-setup@v3
+        with:
+          run_install: false
+
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+
+      - uses: actions/cache@v4
+        name: Setup pnpm cache
+        with:
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: create and publish versions
+        # https://github.com/changesets/action
+        uses: changesets/action@v1
+        with:
+          commit: 'chore: update versions'
+          title: 'chore: update versions'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
will the action fail if I don't provide an npm token in the repo secrets? and if so we actually don't want it to release anything for us, just update the versions and then the user of the starterkit can decide if they do want to publish to npm, we just set-up the functionality to be there in that case you know?

Reply:
If no NPM token is provided, the publish step would fail. However since all packages and apps are set to `private` there won't be any package to be published to npm, so changesets would not even try, and therefor it shouldn't throw an error.

Versioning should remain working as per usual. I will run a test on my formed repo before we merge this.